orthoinference/src/main/java/org/reactome/datafix/KEGGAddLinksMycobacteriumH37RvLinkFix.java

```diff
@@ -0,0 +1,103 @@
+        String database = props.getProperty("release_current.name");
+        String host = props.getProperty("release.database.host");
+        int port = Integer.valueOf(props.getProperty("release.database.port"));
+        long personId = Long.valueOf(props.getProperty("personId"));
+
+        MySQLAdaptor dba = new MySQLAdaptor(host, database, username, password, port);
+
+        Map<String, String> identifierToPrefixMap = new HashMap<>();
+
+        GKInstance instanceEditInst = InstanceEditUtils.createInstanceEdit(dba, personId, "org.reactome.release");
+        instanceEditInst.addAttributeValue(ReactomeJavaConstants.note, "In release 71, changed accessUrl in 'KEGG Gene M. tuberculosis H37Rv' refDB and the corresponding identifier in its referrals");
+        dba.updateInstance(instanceEditInst);
+
+        // mtv-mappings.txt is the result of grepping all uniprot files for 'mtv:'. Same was done for 'mtu:'.
+        // Each of these lines (there are 178) maps to one of the identifiers we want to update.
+        // Here I iterate through each line in each file, and put each identifier in a map that looks like <identifierWithoutPrefix, identifierWithPrefix>
+        // Warning messages are informal, and just for fun.
+
+        String strv;
+        BufferedReader brv = new BufferedReader(new FileReader(new File("src/main/resources/mtv-mappings.txt")));
+        while ((strv = brv.readLine()) != null) {
+            String prefixWithIdentifier = strv.split("\t")[1];
+            String identifier = prefixWithIdentifier.split(":")[1];
+            if (identifierToPrefixMap.get(identifier) == null) {
+                identifierToPrefixMap.put(identifier, prefixWithIdentifier);
+            } else {
+                System.out.println("MAYDAY!");
+            }
+        }
+
+        String stru;
+        BufferedReader bru = new BufferedReader(new FileReader(new File("src/main/resources/mtu-mappings.txt")));
+        while ((stru = bru.readLine()) != null) {
+            String prefixWithIdentifier = stru.split("\t")[1];
+            String identifier = prefixWithIdentifier.split(":")[1];
+            if (identifierToPrefixMap.get(identifier) == null) {
+                identifierToPrefixMap.put(identifier, prefixWithIdentifier);
+            } else {
+                System.out.println("LOOKOUT!");
+            }
+        }
+
+        //TODO: Should display name be updated? Technically the identifier includes the prefix, but from Reactome's perspective we've never included the prefix for any of the KEGG identifiers.
+
+        // Now with the map, I get the reference database of interest. Then I get all referrals of it, which corresponds to the 178 DatabaseIdentifiers.
+        GKInstance keggMycoDBInst = (GKInstance) dba.fetchInstanceByAttribute(ReactomeJavaConstants.ReferenceDatabase, ReactomeJavaConstants.name, "=", "KEGG Gene (Mycobacterium tuberculosis H37Rv)").iterator().next();
```

Comment:
There's only one KEGG Gene (Mycobacterium tuberculosis) reference database in release_current, right?

Reply:
Yes there is only one for H37Rv, and another for another strain called CDC1551

Reply:
Ah okay and the affected instances are all for H37Rv?

Reply:
Yes they are, sorry for belated answer. There was 178 DatabaseIdentifiers that are for H37Rv ReferenceDatabase (determined by solomon) and that was the exact number of referrals to the ReferenceDatabase that were updated